### DISCUSSION TITLE
Report shared version which shouldn't be shared

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [renovatebot/renovate#26213] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

I don't know if this is already happening, but after doing a quick look in the docs I would like to ask:

Can a warning be shown in the dependency board and the pull requests if a version is shared with multiple artifacts? The use case would be that it shouldn't as it was a causality all versions exist but it is not the publisher's intention.

I have been moving to use the shared version plus `minimumReleaseAge`. But I am not sure if all versions that I think are shared are indeed shared plus another use case in which in the future they can be changed to stop being shared. For example https://github.com/kotest/kotest/issues/3486.

**Comment by rarkins** [comment]
It's a good idea, but I'm not sure that we have this level of visibility to make the assessment today. Today for applicable managers we modify "groupName" when a variable is present - I would prefer to elevate variableName as a first-class concept and then maybe we can add more intelligence around it

**Reply by JavierSegoviaCordoba** [reply]
Additionally, what does Renovate if a version that doesn't exist in any source is present? Is there a warning?

I haven't checked it properly but when fixing catalogs of dependencies to start/stop using shared versions, it can be problematic if a non-existent version is set, manually.

Obviously, and in general, it shouldn't be a problem as the consumer will fail when tries to use it, but my use case is a repository that has a catalog of dependencies that are consumed in other repositories, and sometimes I add some new dependencies for future projects but they are not yet consumed.

**Reply by rarkins** [reply]
Please stick to one topic per discussion in future, but the answer is that there's a debug message when the current version can't be found: https://github.com/renovatebot/renovate/blob/f7bde82f662e8e2852daea2835ced40a5920237c/lib/workers/repository/process/lookup/index.ts#L199-L209
